src/deepqmc/wf/paulinet/omni.py

```diff
@@ -1,158 +1,161 @@
-        with_r_backflow (bool): whether real-space backflow is used
-        schnet_kwargs (dict): extra arguments passed to :class:`ElectronicSchNet`
-        subnet_kwargs (dict): extra arguments passed to :class:`SubnetFactory`
-
-    Shape:
-        - :meth:`forward_jastrow`:
-            - Input1, :math:`\mathbf e(\lvert\mathbf r_i-\mathbf r_j\rvert)`:
-              :math:`(*,N,N,\dim(\mathbf e))`
-            - Input2, :math:`\mathbf e(\lvert\mathbf r_i-\mathbf R_I\rvert)`:
-              :math:`(*,N,M,\dim(\mathbf e))`
-            - Output, :math:`J`: :math:`(*)`
-        - :meth:`forward_backflow`:
-            - Input1, :math:`\mathbf e(\lvert\mathbf r_i-\mathbf r_j\rvert)`:
-              :math:`(*,N,N,\dim(\mathbf e))`
-            - Input2, :math:`\mathbf e(\lvert\mathbf r_i-\mathbf R_I\rvert)`:
-              :math:`(*,N,M,\dim(\mathbf e))`
-            - Output, :math:`\kappa_{q\mu i}`:
-              :math:`(*,C,N,N_\text{orb})`
-
-    Attributes:
-        schnet: :class:`ElectronicSchNet` network
-        jastrow: :class:`torch.nn.Sequential` representing a vanilla DNN
-        backflow: :class:`torch.nn.Sequential` representing a vanilla DNN
-    """
 
+class OmniCore(nn.Module):
     def __init__(
         self,
         mol,
-        dist_feat_dim,
         n_up,
         n_down,
-        n_orbitals,
-        n_channels,
+        backflow_spec,
         *,
-        embedding_dim=128,
-        with_jastrow=True,
-        n_jastrow_layers=3,
-        with_backflow=True,
-        n_backflow_layers=3,
-        with_r_backflow=False,
-        schnet_kwargs=None,
-        subnet_kwargs=None,
+        dist_feat_dim=32,
+        dist_feat_cutoff=10.0,
+        many_body_dnn_factory=ElectronicSchNet,
+        mean_field_dnn_factory=partial(ElectronicSchNet, version='mean-field'),
+        jastrow_factory=Jastrow,
+        jastrow_stream='many_body_emb',
+        backflow_factory=Backflow,
+        backflow_stream='many_body_emb',
+        many_body_dnn_kwargs={},
+        mean_field_dnn_kwargs={},
+        jastrow_kwargs={},
+        backflow_kwargs={},
     ):
+
         super().__init__()
-        self.schnet = ElectronicSchNet(
-            n_up,
-            n_down,
-            len(mol),
-            dist_feat_dim=dist_feat_dim,
-            embedding_dim=embedding_dim,
-            subnet_metafactory=partial(SubnetFactory, **(subnet_kwargs or {})),
-            **(schnet_kwargs or {}),
-        )
-        if with_jastrow:
-            self.jastrow = get_log_dnn(embedding_dim, 1, SSP, n_layers=n_jastrow_layers)
-        else:
-            self.forward_jastrow = None
-        if with_backflow:
-            backflow = [
-                get_log_dnn(
-                    embedding_dim,
-                    n_orbitals,
-                    SSP,
-                    n_layers=n_backflow_layers,
-                    last_bias=True,
+        self.dnn_core = nn.ModuleDict(
```

Comment:
The dict is never used as a whole, it's always just its components, so having two separate attributes would be more straightforward.